docs/cli/ilab-model-serve.md

```diff
@@ -0,0 +1,103 @@
+# Design for `ilab model serve`
+
+## Background
+
+With the [request from the community](https://github.com/instructlab/instructlab/issues/1106) for `ilab` to serve different backends such as [vllm](https://docs.vllm.ai/en/stable/) and the [cli redesign](https://github.com/instructlab/dev-docs/pull/48), this design doc's purpose is to flesh out the behavior of the `ilab model serve` command.
+
+Specifically the design of subcommand's of `ilab model serve` that apply to different serving backends.
+
+## Design
+
+Since the subject of the `ilab model serve` command is a model, regardless of the format of the model, every command takes in the `--model` flag or uses its default value in the config.
+
+`ilab model serve` has a subcommand for each backend it can serve models with. As of this draft, the two backends `ilab` would serve with are `llama-cpp` and `vllm`.
+
+This would lead to the commands:
+
+- `ilab model serve llama-cpp`
+- `ilab model serve vllm`
+
+There are specific flags for `ilab model serve` that would apply to all backends. These can be viewed by running `ilab model serve --help`. 
+For the specific subcommands, each has their own flags that can be viewed via `ilab model serve <BACKEND> --help`.
+This ensures there is no mix up of flags between backends.
+
+The following is an overview for the flags of `ilab model serve`:
+
+```console
+ilab model serve
+|
+|_______ --model
+|
+|_______ (other backend agnostic flags)
+|
+|_______ llama-cpp
+|        |
+|        |_______ --num-gpu-layers
+|        |
+|        |_______ --max-ctx-size
+|        |
+|        |_______ (other llama-cpp specific args)
+|
+|_______ vllm
+|        |
+|        |_______ (popular vllm-specific specific args)
+|        |
+|        |_______ --vllm-args
+```
+
+## Command Examples
+
+### Bare-bones but model specific command
+
+```shell
+ilab model serve --model <PATH> 
+```
+
+- Serves the model at `<PATH>`.
+- If the `<PATH>` is the path for a model that can be run by `llama-cpp` then `llama-cpp` is automatically used as the model serving backend.
+- If the `<PATH>` is the path for a model that can be run by `vllm` then `vllm` is automatically used as the model serving backend.
+- If the model at `<PATH>` can be run by either backend, then a default backend defined in the config will be used as the model serving backend.
```

Comment:
What if no backend is specified in the config?

how would express this config option? Config options and flags are the same thing. A config item here implies “—backend <foo>”

Reply:
This another reason  I think we should take another look at just making this an option (consistency between config and flags)

```suggestion
- If the model at `<PATH>` can be run by either backend, then a default backend defined in the config will be used as the model serving backend. In the case where there is ambiguity and a setting is not defined, a hardcoded preference will be used (all currently supported providers do not have this issue). A future profile specification will likely replace the hardcoded fallback. 
```